medusa/metadata/generic.py

```diff
@@ -897,6 +824,55 @@ def _season_banners_dict(self, show_obj, season):
 
         return result
 
+    def _get_show_data(self, show_obj):
+        """Retrieve show data from the indexer.
+
+        Try to reuse the indexer_api class instance attribute.
+        As whe are doing reusing the indexers results, we need to do a full index including actors and images.
+
+        :param show_obj: A TVshow object.
+        :return: A re-indexed show object.
+        """
+
+        show_id = show_obj.indexerid
+
+        indexer_lang = show_obj.lang
+
+        try:
+            if not self.indexer_api:
+                l_indexer_api_params = indexerApi(show_obj.indexer).api_params.copy()
+                l_indexer_api_params['banners'] = True
+                l_indexer_api_params['actors'] = True
+
+                if indexer_lang and not indexer_lang == app.INDEXER_DEFAULT_LANGUAGE:
+                    l_indexer_api_params['language'] = indexer_lang
+
+                if show_obj.dvdorder != 0:
+                    l_indexer_api_params['dvdorder'] = True
+
+                self.indexer_api = indexerApi(show_obj.indexer).indexer(**l_indexer_api_params)
+
+            my_show = self.indexer_api[int(show_id)]
+        except IndexerShowNotFound:
+            logger.log(u'Unable to find {indexer} show {id}, skipping it'.format
+                       (indexer=indexerApi(show_obj.indexer).name,
+                        id=show_id), logger.WARNING)
+            return False
+
+        except (IndexerError, RequestException):
```

Comment:
IndexerUnavailable is handled by IndexerError. IndexerShowIncomplete is too specific. We could choose to replace all with IndexerException?